examples/apply/apply_bp/apply_bp.jl

```diff
@@ -92,29 +89,28 @@ function simple_update_bp(
 
       # Update message tensor
       ψψ = norm_network(ψ)
-      mts[s1] = ITensorNetwork(dictionary([(v1, 1) => ψψ[v1, 1], (v1, 2) => ψψ[v1, 2]]))
-      mts[s2] = ITensorNetwork(dictionary([(v2, 1) => ψψ[v2, 1], (v2, 2) => ψψ[v2, 2]]))
-      mts[s1 => s2] = ITensorNetwork(obs.singular_values)
-      mts[s2 => s1] = ITensorNetwork(obs.singular_values)
+      pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
+      mts[pe] = dense.(obs.singular_values)
+      mts[reverse(pe)] = dense.(obs.singular_values)
     end
     if regauge
       println("regauge")
       mts = belief_propagation(
-        ψψ, mts; contract_kwargs=(; alg="exact"), niters=50, target_precision=1e-5
+        pψψ, mts; contract_kwargs=(; alg="exact"), niters=50, target_precision=1e-5
       )
     end
   end
-  return ψ, mts
+  return ψ, pψψ, mts
```

Comment:
It feels a bit strange to me to output `ψ`, `pψψ`, and `mts`, since there is a lot of overlapping information across those objects. For example, `ψ` is already stored in `pψψ`, both `pψψ` and `mts` store some information about the graph partitioning.

I think this is worth thinking about more carefully, since it has implications for how to design the BP-based `alternating_update` code.